client/src/utilities/column.js

```diff
@@ -97,3 +97,22 @@ export const columnSelectOptions = (intl, columns) => {
   const res = extractColumnOptions(columns);
   return isImmutable(res) ? res.toArray() : res;
 };
+
+// For combine columns and generate geopoints column select
+export const columnBasedOnOppositeColumn = (columns, oppositeColumnName) => {
```

Comment:
perhaps we could find a better name for this function 🤔 . The logic inside doesn't seem to be represented by the name
My proposal and refactor will be:
```js
export const filterColumnsByDataGroupDimension = (columns, columnName) => {
  // if column is not selected the return all columns
  if (!columnName) {
    return columns;
  }

  const column = columns.find(col => col.get('columnName') === columnName);

  // if column is not repeatable return all not repeatable columns
  if (!column.get('repeatable')) {
    return columns.filter(col => !col.get('repeatable'));
  }

  // if column is repeatable, return only columns from the same group
  return columns.filter(col => col.get('groupName') === column.get('groupName'));
};

```